src/platforms/ctxlink/platform.c

```diff
@@ -245,6 +374,50 @@ void platform_target_clk_output_enable(bool enable)
 	(void)enable;
 }
 
+const char *platform_battery_voltage(void)
+{
+	static char ret[64] = {0};
+	if (battery_present == true) {
+		uint32_t battery_voltage = input_voltages[CTXLINK_ADC_BATTERY];
+		//
+		// Format the return string, this will be appened to the target voltage string
+		//
+		strcpy(ret, "\n      Battery : ");
+		uint32_t append_index = strlen(ret);
+		ret[append_index++] = '0' + battery_voltage / 1000U;
+		ret[append_index++] = '.';
+		battery_voltage = battery_voltage % 1000;
+		ret[append_index++] = '0' + battery_voltage / 100U;
+		battery_voltage = battery_voltage % 100;
+		ret[append_index++] = '0' + battery_voltage / 10U;
+		ret[append_index++] = 'V';
+		ret[append_index++] = '\n';
+		ret[append_index] = 0x00;
+	} else
+		memcpy(ret, "\n      Battery : Not present", strlen("\n      Battery : Not present") + 1);
+	return ret;
+}
+
+bool platform_check_battery_voltage(void)
+{
+	platform_adc_read();
+	//
+	// Is battery connected?
+	//
+	if ((input_voltages[CTXLINK_ADC_BATTERY] <= BATTERY_VOLTAGE_1) ||
+		(input_voltages[CTXLINK_ADC_BATTERY] >= BATTERY_VOLTAGE_2)) {
```

Comment:
The extra parens around the two conditions are superfluous and can be dropped.

Reply:
Fixed